src/main/java/io/debezium/connector/vitess/VitessStreamingChangeEventSource.java

```diff
@@ -94,16 +94,25 @@ public void execute(ChangeEventSourceContext context, VitessPartition partition,
 
     private ReplicationMessageProcessor newReplicationMessageProcessor(VitessPartition partition,
                                                                        VitessOffsetContext offsetContext) {
-        return (message, newVgtid, isLastRowOfTransaction) -> {
+        return (message, newVgtid) -> {
             if (message.isTransactionalMessage()) {
-                // Tx BEGIN/END event
-                offsetContext.rotateVgtid(newVgtid, message.getCommitTime());
```

Comment:
In the old code, rotateVgtid() is called on all cases of isTransactionMessage() (including BEGIN and COMMIT and other unknown types), do we need to cover that unknown types in the new code?

Reply:
It might be safer to add an else {assert here} in case our assumptions are not correct.

Reply:
I added validation when transactional messages are created that they are only commit or begin. Changed to else since these are the only two cases. And add unit tests for verifying this begin/commit validation